### PR TITLE
Add club checkup

### PR DESCRIPTION
**Pull request by maxwofford**

### CODE REVIEW

**Review by lachlanjc — APPROVED** [review]

File: src/components/club_checkup/LeaderForm.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import { withFormik } from 'formik'
+import yup from 'yup'
+
+import {
```
Comment (lachlanjc) [inline src/components/club_checkup/LeaderForm.js]: This is likely WIP, but this import is empty, and the file doesn't export anything :)

Comment (maxwofford) [reply src/components/club_checkup/LeaderForm.js]: Yeah, the commit called `SAVEPOINT` was just to transfer changes between computers

File: src/components/checkup/ClubForm.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  handleChange,
+  handleBlur,
+  handleSubmit,
+  isSubmitting,
+}) => (
+  <form onSubmit={handleSubmit}>
+    <Field
+      name="high_school_name"
+      value={values.high_school_name}
+      label="School Name"
```
Comment (lachlanjc) [inline src/components/checkup/ClubForm.js]: Style thing, we only capitalize the first character of form labels (except proper nouns)

**Comment by lachlanjc** [comment]
This isn't really relevant to this PR, but—

We really should redo how we do forms in our codebase. Since I mostly work on our small forms, it hasn't become a priority to me, but it grows rather unwieldy with this many fields. The temporary solution is a one-off [`field` helper](https://github.com/hackhv/site/blob/ef9f9c1c87a4a1d07b2f107a8ab5ae1b23557d2f/src/components/RegistrationForm.js#L49), but the best solution would combine [Formik's `<Field />` component](https://github.com/jaredpalmer/formik#render-props-formik--and-field) with the DS `<Field />`. On a form like this one, the components are incredibly messy vs. Rails. Not sure when I'll get to this, but it should definitely happen pre-Community.

**Comment by lachlanjc** [comment]
Relevant to this PR, let's continue using our form style of having only the first character + proper nouns + acronyms capitalized in form labels. ("School Name", "LinkedIn Profile", etc)

File: src/components/checkup/LeaderForm.js [hunk 3]
```diff
@@ -230,92 +213,84 @@ const InnerForm = ({
...
-    new_leader_phone_number: props.new_leader.phone_number || '',
-    new_leader_twitter_url: props.new_leader.twitter_url || ''
-  }),
+  mapPropsToValues: props => {
+    const values = {}
+    const combinedProps = {
+      ...props.new_leader,
+      email: props.email
+    }
+    Object.keys(combinedProps).forEach(key => {
```
Comment (lachlanjc) [inline src/components/checkup/LeaderForm.js:222]: yesssss

**Pull request merged**